src/main/kotlin/com/openlattice/chronicle/jobs/BackgroundChronicleDeletionService.kt

```diff
@@ -0,0 +1,73 @@
+package com.openlattice.chronicle.jobs
+
+import com.openlattice.chronicle.auditing.AuditedOperationBuilder
+import com.openlattice.chronicle.auditing.AuditingManager
+import com.openlattice.chronicle.postgres.ResultSetAdapters
+import com.openlattice.chronicle.services.jobs.JobService
+import com.openlattice.chronicle.storage.ChroniclePostgresTables.Companion.JOBS
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.CREATED_AT
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.JOB_DATA
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.JOB_ID
+import com.openlattice.chronicle.storage.StorageResolver
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.annotation.Scheduled
+import java.sql.Connection
+
+/**
+ * @author Solomon Tang <solomon@openlattice.com>
+ */
+class BackgroundChronicleDeletionService(
+  private val jobService: JobService,
+  private val storageResolver: StorageResolver,
+  private val auditingManager: AuditingManager,
+) {
+    companion object {
+        private val logger = LoggerFactory.getLogger(BackgroundChronicleDeletionService::class.java)!!
+
+        private val GET_NEXT_JOB_SQL = """
+            DELETE FROM ${JOBS.name}
+            WHERE ${JOB_ID.name} = (
+                SELECT ${JOB_ID.name}
+                FROM ${JOBS.name}
+                WHERE ${JOB_DATA.name} ->> '@type' = '${DeleteStudyUsageData::class.java}'
+                ORDER BY ${CREATED_AT.name} ASC
+                FOR UPDATE SKIP LOCKED
+                LIMIT 1
+            )
+            RETURNING *
+        """.trimIndent()
+    }
+
+    @Scheduled(fixedRate = 30_000L)
+    fun deleteChronicleUsageData() {
```

Comment:
(1) You need to rate limit the maximum number of jobs. 
(2) @Scheduled executes in a fixed size shared thread pool, so long running jobs have a risk of creating a backlog.
(3) Since postgres is handling making sure that multiple threads are working on the same job I would have a class level semaphore with a fixed number of permits, I would try to acquire a permit, if successful hand off to a fix sized threadpool executor.